ufl/split_functions.py

```diff
@@ -85,8 +89,18 @@ def split(v):
     # Build expressions representing the subfunction of v for each subelement
     offset = begin
     sub_functions = []
-    domains = domain.iterable_like(element)
-    for i, (d, e) in enumerate(zip(domains, element.sub_elements)):
+
+    # Deal with symmetry
+    if isinstance(element.pullback, SymmetricPullback):
+        symmetry = element.pullback._symmetry
+        indices = (comp for idx, comp in sorted(symmetry.items()))
+    else:
+        indices = range(len(element.sub_elements))
+
+    domains = tuple(domain.iterable_like(element))
+    for i in indices:
+        d = domains[i]
+        e = element.sub_elements[i]
```

Comment:
Should `SymmetricElement.sub_elements` be returning the tuple with the symmetries applied or only the linearly independent ones?